pkg/crd/testdata/cronjob/v2/cronjob_types.go

```diff
@@ -0,0 +1,236 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO(directxman12): test this across both versions (right now we're just
+// trusting k/k conversion, which is probably fine though)
+
+// +groupName=testdata.kubebuilder.io
+// +versionName=v2
+package v2
+
+import (
+	"fmt"
+
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CronJobSpec defines the desired state of CronJob
+type CronJobSpec struct {
+	// V2Field is a dummy field that does not exist in v1.
+	V2Field string `json:"v2field"`
```

Comment:
Differences with `v1/cronjob_types`:

- set package name to `v2`
- add `V2Field` to make sure the target CRD is different
- set `kubebuilder:storageversion`